pyquil/api/_qam.py

```diff
@@ -90,6 +75,33 @@ def get_register_map(self) -> Dict[str, Optional[np.ndarray]]:
         register_map = self.data.result_data.to_register_map()
         return {key: matrix.to_ndarray() for key, matrix in register_map.items()}
 
+    def get_raw_readout_data(self) -> Union[RawQVMReadoutData, RawQPUReadoutData]:
+        """
+        Get the raw result data. This will be a flattened structure derived
+        from :class:`qcs_sdk.qvm.QVMResultData` or :class:`qcs_sdk.qpu.QPUResultData`
+        depending on where the job was run. See their respective documentation
+        for more information on the data format.
+
+        This property should be used when running programs that use features like
+        mid-circuit measurement and dynamic control flow on a QPU, since they can
+        produce irregular result shapes that don't necessarily fit in a
+        rectangular matrix. If the program was run on a QVM, or doesn't use those
+        features, consider using the ``register_map`` property instead.
+        """
+        return self.data.result_data.to_raw_readout_data()
+
+    def get_memory_values(self) -> Mapping[str, Optional[MemoryValues]]:
```

Comment:
Note depending on how the discussion on https://github.com/rigetti/qcs-sdk-rust/pull/393 shakes out to possibly update the return type